Mathlib/Probability/Kernel/Condexp.lean

```diff
@@ -71,91 +61,132 @@ end AuxLemmas
 
 variable {Ω F : Type _} [TopologicalSpace Ω] {m : MeasurableSpace Ω} [mΩ : MeasurableSpace Ω]
   [PolishSpace Ω] [BorelSpace Ω] [Nonempty Ω] {μ : Measure Ω} [IsFiniteMeasure μ]
-  [NormedAddCommGroup F] {f : Ω → F}
 
 /-- Kernel associated with the conditional expectation with respect to a σ-algebra. It satisfies
 `μ[f | m] =ᵐ[μ] fun ω => ∫ y, f y ∂(condexpKernel μ m ω)`.
 It is defined as the conditional distribution of the identity given the identity, where the second
-identity is understood as a map from `Ω` with the σ-algebra `mΩ` to `Ω` with σ-algebra `m`. -/
+identity is understood as a map from `Ω` with the σ-algebra `mΩ` to `Ω` with σ-algebra `m ⊓ mΩ`.
+We use `m ⊓ mΩ` instead of `m` to ensure that it is a sub-σ-algebra of `mΩ`. -/
 noncomputable irreducible_def condexpKernel (μ : Measure Ω) [IsFiniteMeasure μ]
-    (m : MeasurableSpace Ω) : @kernel Ω Ω m mΩ :=
-  @condDistrib Ω Ω Ω _ mΩ _ _ _ mΩ m id id μ _
+    (m : MeasurableSpace Ω) : @kernel Ω Ω (m ⊓ mΩ) mΩ :=
```

Comment:
Can't you reinterpret it as a kernel from `m` to `mΩ`? (So that it has the right type `@kernel Ω Ω m mΩ` from the start, and the user never needs to worry about `m ⊓ mΩ`?)

Reply:
I added a `kernel.comap` to the definition to turn it into a kernel from `m` to `mΩ`.